### PR TITLE
Yt import tb

### LINKED ISSUE #28: Import YouTube playlist
Currently tutorials can be created by manually entering information and then manually entering videos one by one.

The playlist importer should take a playlist_id which relates to a youtube playlist and import all of the videos and associate them with a tutorial.

```
As an admin
When I visit '/admin/tutorials/new'
Then I should see a link for 'Import YouTube Playlist'
When I click 'Import YouTube Playlist'
Then I should see a form

And when I fill in 'Playlist ID' with a valid ID
Then I should be on '/admin/dashboard'
And I should see a flash message that says 'Successfully created tutorial. View it here.'
And 'View it here' should be a link to '/tutorials/:id'
And when I click on 'View it here'
Then I should be on '/tutorials/:id'
And I should see all videos from the YouTube playlist
And the order should be the same as it was on YouTube
```

- [x] Retain the original sequence from YouTube
- [x] Creates a video record for each video in the playlist and stores the youtube id.
- [x] Associates each video stored in the DB is associated with a tutorial. (validate presence of tutorial id)
- [x] Can import a playlist with more than 50 videos

### PR DESCRIPTION
**Pull request by TravisBorgsmiller**
Adds functionality to allow admin to create new playlists through a form that takes in a youtube playlist id. 

New files:

views/admin/tutorials/new_import << new form for creating playlist
models/you_tube/playlist << poro for formatting data to create tutorial objects
spec/features/admin/can_import_playlist_spec.rb
Includes tests for all listed edge cases
Other:

Rubocop has 2 citations

### CODE REVIEW

File: app/controllers/admin/tutorials_controller.rb [hunk 1]
```diff
@@ -3,10 +3,18 @@ def edit
...
   end
 
-  def create; end
+  def create
+    if tutorial_params[:playlist_id]
+      tutorial = Tutorial.import_from_playlist(tutorial_params[:playlist_id])
+      flash[:success] = "Successfully created tutorial.\
+       #{view_context.link_to 'View it here', tutorial_path(tutorial)}."
+      redirect_to admin_dashboard_path
+    end
```
Comment (alex-latham) [inline app/controllers/admin/tutorials_controller.rb]: I just learned we can fix one of gthe rubocop citations by replacing this section with (blank line included):
```
    return unless tutorial_params[:playlist_id]

    tutorial = Tutorial.import_from_playlist(tutorial_params[:playlist_id])
    flash[:success] = "Successfully created tutorial. \
      #{view_context.link_to 'View it here', tutorial_path(tutorial)}."
    redirect_to admin_dashboard_path

```
The `return` line prevents the following lines from executing unless a playlist id was sent with the tutorial params.

File: app/models/tutorial.rb [hunk 2]
```diff
@@ -2,4 +2,26 @@ class Tutorial < ApplicationRecord
...
+      Video.create(
+        title: yt_video.title,
+        description: yt_video.description,
+        thumbnail: yt_video.thumbnail,
+        video_id: yt_video.id,
+        tutorial: tutorial
+      )
+    end
+    tutorial
+  end
```
Comment (alex-latham) [inline app/models/tutorial.rb:26]: Helper method prime candidate

File: app/services/you_tube_service.rb [hunk 3]
```diff
@@ -5,6 +5,18 @@ def video_info(id)
...
+    params = { part: 'snippet', id: playlist_id }
+
+    get_json('youtube/v3/playlists', params)
+  end
+
+  def playlist_items(playlist_id)
+    params = { part: 'snippet', maxResults: 50, playlistId: playlist_id }
+
+    get_json('youtube/v3/playlistItems', params)
+  end
```
Comment (alex-latham) [inline app/services/you_tube_service.rb:18]: We'll need to expand on this method at some point in order to make additional API calls for paginated results (results > 50). Probably another place we can use `return unless ...`!

File: app/views/admin/tutorials/new_import.html.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<h2>Import Youtube Playlist</h2>
```
Comment (alex-latham) [inline app/views/admin/tutorials/new_import.html.erb:1]: Small typo - capital T in YouTube

File: db/schema.rb [hunk 5]
```diff
@@ -67,6 +67,8 @@
     t.integer "role", default: 0
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.integer "uid"
+    t.string "github_token"
```
Comment (alex-latham) [inline db/schema.rb:71]: What's `uid`?

File: db/schema.rb [hunk 6]
```diff
@@ -67,6 +67,8 @@
...
     t.string "gh_token"
```
Comment (alex-latham) [inline db/schema.rb:72]: Is `uid` for the oauth story? Also, looks like we picked up an extra `github_token`. I would drop the first one since our app currently looks for the token in `gh_token`
